### PR TITLE
Refactor transactions confirmations webhook

### PR DESCRIPTION
**Pull request by axelsheva**

### CODE REVIEW

File: src/shared/emitter.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+export interface IBaseEmitter<EVENT> {
+    on(event: EVENT, fn: Function): this;
+    emit<T>(event: EVENT, ...args: Array<T>): this;
```
Comment (user-dob) [inline src/shared/emitter.ts]: if I want to pass different types?

File: src/shared/emitter.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        if (this.listeners.has(event)) {
+            this.listeners.get(event).forEach(fn => fn(...args));
+        }
+
+        return this;
+    }
+
+    removeListener(event: EVENT, fn: Function): this {
+        if (this.listeners.has(event)) {
+            this.listeners.get(event).filter(f => f !== fn);
```
Comment (user-dob) [inline src/shared/emitter.ts]: result after filter not save

Comment (user-dob) [reply src/shared/emitter.ts]: fn?: Function - in inteface

File: src/service/index.ts [hunk 3]
```diff
@@ -25,21 +25,30 @@ import { Node } from 'src/model/node';
...
+export const transactionService = new TransactionCreator(
     slotService,
     timeService,
     accountService,
 );
+
+const nodes = NODE_HOSTS
+    .map(host => initSocketIOClient(host.ip, host.port))
+    .map(socket => new SocketIOClient<API_ACTION_TYPES | EVENT_TYPES>(socket))
+    .map(socket => new Node(socket));
```
Comment (user-dob) [inline src/service/index.ts]: What is the point?

File: src/shared/emitter.ts (hunk 1 above)
Comment (axelsheva) [reply src/shared/emitter.ts]: You should pass only specified types

File: src/service/index.ts (hunk 3 above)
Comment (axelsheva) [reply src/service/index.ts]: The Node contains information about blockchain

File: src/shared/emitter.ts (hunk 2 above)
Comment (axelsheva) [reply src/shared/emitter.ts]: Fixed

**Pull request merged**